catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java

```diff
@@ -184,21 +182,21 @@ public void executeQuery(final SearchRequest request, final ServerSession sessio
 
                 future.get(timeRemaining, TimeUnit.MILLISECONDS);
             } catch (InterruptedException e) {
-                LOGGER.info("Query interrupted for source " + sourceId, e);
-                failSource(request, session, search, sourceId);
+                LOGGER.info("Query interrupted for source {}", sourceId, e);
+                failSource(request, session, search, sourceId, e);
             } catch (TimeoutException e) {
-                LOGGER.info("Query timed out for source " + sourceId, e);
-                failSource(request, session, search, sourceId);
+                LOGGER.info("Query timed out for source {}", sourceId, e);
+                failSource(request, session, search, sourceId, e);
             } catch (ExecutionException e) {
                 LOGGER.warn("Query failed for source " + sourceId, e.getCause());
-                failSource(request, session, search, sourceId);
+                failSource(request, session, search, sourceId, e);
             }
         }
     }
 
     private void failSource(SearchRequest request, ServerSession session, Search search,
-            String sourceId) {
-        search.failSource(sourceId);
+            String sourceId, Exception cause) {
+        search.failSource(sourceId, cause);
         try {
             pushResults(request.getId(), search.transform(request.getId()), session);
```

Comment:
I know you didn't change this, but do you think `publishResults` or `sendResults` be clearer than `pushResults`? The latter feels like we're just putting the result on a stack, not that we are actually publishing it right away.